src/services/anchor-service.ts

```diff
@@ -647,19 +645,11 @@ export class AnchorService {
    */
   async _buildCandidates(requests: Request[]): Promise<Array<Candidate>> {
     const candidates = []
-    const metadataByStreamId = await this.metadataService
-      .batchRetrieve(requests.map((r) => StreamID.fromString(r.streamId)))
-      .then((metadata) => {
-        return Object.fromEntries(metadata.map((m) => [m.streamId.toString(), m]))
-      })
 
     for (const request of requests) {
       const streamId = StreamID.fromString(request.streamId)
-      const metadata = metadataByStreamId[request.streamId]
-      if (metadata) {
-        const candidate = new Candidate(streamId, request, metadata.metadata)
-        candidates.push(candidate)
-      }
+      const candidate = new Candidate(streamId, request, null as unknown as ICandidateMetadata)
```

Comment:
it's annoying that I can't just easily change `Candidate` to not accept metadata at all, or to skip sorting the leaf nodes entirely.  But that code lives in the anchor-utils package in the js-ceramic repo now, and the CAS repo is still using extremely old versions of the js-ceramic packages.  So rather than go through the headache of first upgrading all the js-ceramic packages in the CAS repo, and then updating the anchor-utils package with new versions of the merkle tree builder apis, and then coming back to the CAS to update it to use the updated anchor-utils, I took the shortcut of leaving all the code for building metadata and sorting leaf nodes in place in the underlying code for building merkle trees, I just don't pass a function for building metadata and provide a very simplified leaf sorter.

Reply:
That's an immensely helpful comment!

Any plans to change js-ceramic counterpart?

Reply:
Currently no, not planning to update the types in js-ceramic.  We certainly could get a lot of code cleanup if we do, but in order to use those types here we have to first update the CAS to newer versions of all of the js-ceramic packages.  Might very well be worth doing if we wind up doing much more work on the CAS, but since our current plan is to wrap up CAS scaling work soon and then not touch the CAS anymore until we kill it entirely in favor of self-anchoring, my hope is we can just ignore this code until it goes away.

Happy to revisit that though if this old messy code starts getting in the way of our ability to move fast on any of the projects we need in the future.